asllib/types.ml

```diff
@@ -327,22 +377,27 @@ module Domain = struct
 
   let compare _d1 _d2 = assert false
 
-  let int_set_is_subset is1 is2 =
+  let syntax_is_subset env is1 is2 =
+    (* TODO: improve for basic cases. *)
+    constraints_equal env is1 is2
```

Comment:
perhaps at least check if the sets of constraints are equal instead of list equality?